remove Cylc 7 compatibility mode (#7325)
cylc/cylc-flow: changes.d/6849.break.md

> Cylc 7 compatibility mode has been removed, any workflows defined by a `suite.rc` file must now be upgraded to `flow.cylc` to run with the latest version of Cylc.

Reviewer: (Minor)
```suggestion
Cylc 7 compatibility mode has been removed. Any workflows defined by a `suite.rc` file must now be renamed to `flow.cylc` to run with this version of Cylc.
```
